identity-wallet/src/state/reducers/history.rs

```diff
@@ -0,0 +1,31 @@
+use serde::{Deserialize, Serialize};
+use ts_rs::TS;
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq, TS)]
+#[ts(export)]
+pub enum EventType {
+    #[serde(rename = "added_connection")]
+    AddedConnection,
+    #[serde(rename = "added_credentials")]
+    AddedCredentials,
+    #[serde(rename = "shared_credentials")]
+    SharedCredentials,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq, TS)]
+#[ts(export)]
+pub struct HistoryEvent {
+    pub issuer_name: String,
+    pub event_type: EventType,
+    pub connection_id: Option<String>,
+    pub date: String,
+    pub credentials: Vec<HistoryCredential>,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq, TS)]
+#[ts(export)]
+pub struct HistoryCredential {
```

Comment:
Do you think it makes sense to add the credential id as field so we could directly jump to that credential when scrolling through the history?

Reply:
Hmm yes seems like a good idea, I will add the field to it.